_data-prepper/pipelines/configuration/sources/dynamo-db.md

```diff
@@ -88,8 +88,96 @@ Option | Required | Type | Description
 :--- | :--- | :--- | :---
 `start_position` | No | String | The position from where the source starts reading stream events when the DynamoDB stream option is enabled. `LATEST` starts reading events from the most recent stream record. 
 
+## Exposed metadata attributes
+
+The following metadata will be added to each Event that is processed by the DynamoDB source. These metadata attributes can be accessed with
+the [expression syntax getMetadata function](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/#getmetadata).
+
+* `primary_key` - The primary key of the DynamoDB item. For tables that only contain a partition key, this value will give the partition key. For tables that contain both a partition and sort key, the primary_key attribute will be equal to the partition and sort key, separated by a `|` (i.e. `partition_key|sort_key`)
+* `partition_key` - The partition key of the DynamoDB item
+* `sort_key` - The sort key of the DynamoDB item. This will be null if the table does not have a sort key
+* `dynamodb_timestamp` - The timestamp of the DynamoDB item. This will be the export time for export items, and the DDB stream Event time for stream items. This timestamp is used by sinks to emit an `EndtoEndLatency` metric for DDB stream Events, which tracks the latency between a change in the DDB table, and that change getting applied to the sink.
+* `document_version` - Based off the `dynamodb_timestamp`, and modified to break ties between stream items that are received in the same second. Recommend for use with the `opensearch` sink `document_version` setting.
+* `opensearch_action` - A default value for mapping DDB Event actions to OpenSearch actions. This action will be `index` for export items, and INSERT or MODIFY stream events. This action will be `delete` for REMOVE stream events.
+* `dynamodb_event_name` - The exact event type for the item. Will be `null` for export items, and either `INSERT`, `MODIFY`, or `REMOVE` for stream events.
+* `table_name` - The DynamoDB table name that an event came from.
+
+
+## Permissions
+
+The following shows the minimum required permissions for running DDB as a source
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "allowRunExportJob",
+            "Effect": "Allow",
+            "Action": [
+                "dynamodb:DescribeTable",
```

Comment:
Would the user need `DescribeTable` when using only streams? If so, maybe we could split this out. See my comment below about hinting to the user about reducing permissions when using only export or only stream.

We could also create a GitHub issue and follow-up with it. It needn't hold up the PR if it will take some effort.

Reply:
I just checked the code and looks like DescribeTable is required for streams as well